src/App.tsx

```diff
@@ -1,173 +1,92 @@
-                    <a
-                      className="has-text-danger"
-                      href="#/people/lievijne-jans-1542"
-                    >
-                      Lievijne Jans
-                    </a>
-                  </td>
-
-                  <td>f</td>
-                  <td>1542</td>
-                  <td>1582</td>
-                  <td>-</td>
-                  <td>-</td>
-                </tr>
-
-                <tr data-cy="person">
-                  <td>
-                    <a href="#/people/bernardus-de-causmaecker-1721">
-                      Bernardus de Causmaecker
-                    </a>
-                  </td>
+import { Loader } from './components/Loader';
+import { TableList } from './components/Tabledata.tsx/TableList';
+import { Error } from './components/Error/Error';
+import { Navbar } from './components/Navbar/Navbar';
+import { Person } from './types/Person';
 
-                  <td>m</td>
-                  <td>1721</td>
-                  <td>1789</td>
+import { getPeople } from './api';
 
-                  <td>
-                    <a
-                      className="has-text-danger"
-                      href="#/people/livina-haverbeke-1692"
-                    >
-                      Livina Haverbeke
-                    </a>
-                  </td>
+import './App.scss';
 
-                  <td>
-                    <a href="#/people/lieven-de-causmaecker-1696">
-                      Lieven de Causmaecker
-                    </a>
-                  </td>
-                </tr>
-              </tbody>
-            </table>
+export const App = () => {
+  const [personData, setApiData] = useState<Person[]>([]);
+  const [error, setError] = useState('');
+  const [clickedNav, setClickedNav] = useState(
+    window.location.hash.substr(1) || '/',
+  );
+
+  const location = useLocation();
+
+  const fetchData = () => {
+    getPeople()
+      .then((res) => {
+        setApiData(res);
+      })
+      .catch((err) => {
+        setError(err.message);
+      });
+  };
+
+  useEffect(() => {
+    if (clickedNav === '/') {
+      setApiData([]);
+    }
+
+    fetchData();
+    setClickedNav(window.location.hash.substr(1) || '/');
+  }, [location]);
+
+  useEffect(() => {
+    const handlePopstate = () => {
+      setClickedNav(window.location.hash.substr(1));
+    };
+
+    window.addEventListener('popstate', handlePopstate);
+
+    return () => {
+      window.removeEventListener('popstate', handlePopstate);
+    };
+  }, []);
+
+  const handleNavClick = (route: string) => {
+    window.location.hash = route;
+  };
+
+  return (
+    <HashRouter>
+      <div data-cy="app">
+        <Navbar
+          clickedNavs={clickedNav}
+          handleNavClick={handleNavClick}
+        />
+        <main className="section">
+          <div className="container">
+            <Switch>
+              <Route exact path="/">
+                <h1 className="title">Home Page</h1>
+              </Route>
```

Comment:
would be better to create separate component instead of putting tag directly inside router